pkgs/tools/typesetting/tex/texlive/tl2nix.sed

```diff
@@ -37,38 +37,52 @@ $a}
-    s/"(cc-by(-sa)?-[1-4])"/"\10"/g
-    s/"apache2"/"asl20"/g
-    s/"artistic"/"artistic1-cl8"/g
-    s/"bsd"/"bsd3"/g          # license text does not match exactly, but is pretty close
-    s/"bsd4"/"bsdOriginal"/g
-    s/"collection"/"free"/g   # used for collections of individual packages with distinct licenses. As TeXlive only contains free software, we can use "free" as a catchall
-    s/"fdl"/"fdl13Only"/g
-    s/"gpl"/"gpl1Only"/g
-    s/"gpl([1-3])"/"gpl\1Only"/g
-    s/"gpl2\+"/"gpl2Plus"/g
-    s/"gpl3\+"/"gpl3Plus"/g
-    s/"lgpl"/"lgpl2"/g
-    s/"lgpl2\.1"/"lgpl21"/g
-    s/"lppl"/"lppl13c"/g      # not used consistently, sometimes "lppl" refers to an older version of the license
-    s/"lppl1\.2"/"lppl12"/g
-    s/"lppl1\.3"/"lppl13c"/g  # If a work refers to LPPL 1.3 as its license, this is interpreted as the latest version of the 1.3 license (https://www.latex-project.org/lppl/)
-    s/"lppl1\.3a"/"lppl13a"/g
-    s/"lppl1\.3c"/"lppl13c"/g
-    s/"other-free"/"free"/g
-    s/"opl"/"opubl"/g
-    s/"pd"/"publicDomain"/g
-
-    s/^catalogue-license (.*)/  license = [ \1 ];/p
+  s/^catalogue (.*)/  catalogue = "\1";/p
+
+  /^catalogue-/{
+    :next-cat
+    # extract version and clean unwanted chars from it
+    /^catalogue-version/y/ \/~/_--/
+    /^catalogue-version/s/[\#,:\(\)]//g
+    s/^catalogue-version_(.*)/  version = "\1";/p
+
+    # extract license
+    /^catalogue-license/{
+      # wrap licenses in quotes
+      s/ ([^ ]+)/ "\1"/g
+      # adjust naming as in nixpkgs, the full texts of the licenses are available at https://www.ctan.org/license/${licenseName}
+      s/"(cc-by(-sa)?-[1-4])"/"\10"/g
+      s/"apache2"/"asl20"/g
+      s/"artistic"/"artistic1-cl8"/g
+      s/"bsd"/"bsd3"/g          # license text does not match exactly, but is pretty close
+      s/"bsd4"/"bsdOriginal"/g
+      s/"collection"/"free"/g   # used for collections of individual packages with distinct licenses. As TeXlive only contains free software, we can use "free" as a catchall
+      s/"fdl"/"fdl13Only"/g
+      s/"gpl"/"gpl1Only"/g
+      s/"gpl([1-3])"/"gpl\1Only"/g
+      s/"gpl2\+"/"gpl2Plus"/g
+      s/"gpl3\+"/"gpl3Plus"/g
+      s/"lgpl"/"lgpl2"/g
+      s/"lgpl2\.1"/"lgpl21"/g
+      s/"lppl"/"lppl13c"/g      # not used consistently, sometimes "lppl" refers to an older version of the license
+      s/"lppl1\.2"/"lppl12"/g
+      s/"lppl1\.3"/"lppl13c"/g  # If a work refers to LPPL 1.3 as its license, this is interpreted as the latest version of the 1.3 license (https://www.latex-project.org/lppl/)
+      s/"lppl1\.3a"/"lppl13a"/g
+      s/"lppl1\.3c"/"lppl13c"/g
+      s/"other-free"/"free"/g
+      s/"opl"/"opubl"/g
+      s/"pd"/"publicDomain"/g
+
+      s/^catalogue-license (.*)/  license = [ \1 ];/p
+    }
+
+    s/^.*$//
+    N
+    s/^\ncatalogue-/catalogue-/
+    t next-cat
+    # flag that package has a catalogue entry
+    i\  hasCatalogue = true;
```

Comment:
Can we invert this to write only when it's false? It looks like that would save us some lines in tlpdb.nix.

Reply:
I have been meaning to ask, since it applies to `hasRunfiles = true;` as well, and emitting a line in the *absence* of some data is quite tricky. I can only think of the following:
- more sed dark magic (save the presence of catalogue, runfiles, sha512 of runfiles in the hold space, print the appropriate boolean once we reach the end of each package)
- use nix-instantiate to run a nix function on it
- ignore the issue, it should compress well anyway...

Reply:
We can address separately, if you get a good idea.

Reply:
I switched `hasCalatogue` to default. The same trick can now be used to change `hasRunfiles` and omit `sha512.run` accordingly (not done here, to keep the PR on topic).